lib/msf/core/db_manager/payload.rb

```diff
@@ -9,18 +9,22 @@ def create_payload(opts)
       end
 
       wspace = Msf::Util::DBManager.process_opts_workspace(opts, framework)
-      wspace.payloads.create!(opts)
+      Mdm::Payload.create!(opts)
     end
   end
 
   def payloads(opts)
     ::ActiveRecord::Base.connection_pool.with_connection do
       if opts[:id] && !opts[:id].to_s.empty?
         return Array.wrap(Mdm::Payload.find(opts[:id]))
+      else
+        # Check the database for a matching UUID, returning an empty array if no results are found
+        begin
```

Comment:
good approach! this needs to be robust to failures since it's not going to exist reliably for a while.